internal/cli_utils/datasource_flag.go

```diff
@@ -0,0 +1,31 @@
+package cli_utils
+
+import (
+	"flag"
+	"fmt"
+	"slices"
+)
+
+type datasource_flag struct {
+	Value    string
+	Is_valid bool
+}
+
+func (d *datasource_flag) datasource_binding() {
+	flag.StringVar(&d.Value, "datasource", "fg", "")
+	flag.StringVar(&d.Value, "d", "fg", "")
+}
+
+func (d *datasource_flag) Validation() {
+	/*
+		fg = feed gateway
+		fn = full node rpc
+	*/
+	valid_values := []string{"fg", "fn"}
```

Comment:
therefore we can select case on those two consts to validate user input

Reply:
sure!